### PR TITLE
Add case for domstate

### PR DESCRIPTION
**Pull request by cliping**
1. Add case for check guest's domstate reason after guest core
dumped by 'virsh dump'.
2. Add case for check guest's domstate reason after reboot and
crash guest by 'kill -9'

### CODE REVIEW

**Comment by cliping** [comment]
Test result:

#avocado run --vt-type libvirt --vt-machine-type i440fx virsh.domstate.normal_test.reason.dump_vm
JOB ID     : 027af59660d45090e45b2d2f093677f8e72048ee
JOB LOG    : /root/avocado/job-results/job-2018-01-30T01.48-027af59/job.log
 (1/2) type_specific.io-github-autotest-libvirt.virsh.domstate.normal_test.reason.dump_vm.dump_live: PASS (4.52 s)
 (2/2) type_specific.io-github-autotest-libvirt.virsh.domstate.normal_test.reason.dump_vm.dump_crash: PASS (4.32 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 11.00 s

#avocado run --vt-type libvirt --vt-machine-type i440fx virsh.domstate.normal_test.reason.start_vm
JOB ID     : f9837a927ee905dc7e7f591b602d35088bedff2f
JOB LOG    : /root/avocado/job-results/job-2018-01-30T01.52-f9837a9/job.log
 (1/2) type_specific.io-github-autotest-libvirt.virsh.domstate.normal_test.reason.start_vm.normal: PASS (4.78 s)
 (2/2) type_specific.io-github-autotest-libvirt.virsh.domstate.normal_test.reason.start_vm.rename: PASS (4.26 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 11.20 s

**Comment by cliping** [comment]
Test result:

#avocado run --vt-type libvirt --vt-machine-type i440fx virsh.domstate.error_test.kill_vm
JOB ID     : ba483e5a0f072e7f145d52dcb6a1428120411dab
JOB LOG    : /root/avocado/job-results/job-2018-02-01T01.35-ba483e5/job.log
 (1/3) type_specific.io-github-autotest-libvirt.virsh.domstate.error_test.kill_vm.normal: PASS (4.21 s)
 (2/3) type_specific.io-github-autotest-libvirt.virsh.domstate.error_test.kill_vm.after_stopping_libvirtd: PASS (4.82 s)
 (3/3) type_specific.io-github-autotest-libvirt.virsh.domstate.error_test.kill_vm.after_reboot_vm: PASS (4.19 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 15.42 s

File: libvirt/tests/src/virsh_cmd/monitor/virsh_domstate.py [hunk 1]
```diff
@@ -145,11 +153,21 @@ def run(test, params, env):
                 virsh.destroy(vm_name, ignore_status=False)
             elif vm_action == "start":
                 virsh.destroy(vm_name, ignore_status=False)
-                virsh.start(vm_name, ignore_status=False)
+                if start_action == "rename":
+                    os.rename(image_source, new_image_source)
+                    virsh.start(vm_name, ignore_status=True)
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/monitor/virsh_domstate.py]: line 158 and line 160 both startvm ,you can extract them out:
if start_action:
   os.rename
virsh.start(xx)

Comment (cliping) [reply libvirt/tests/src/virsh_cmd/monitor/virsh_domstate.py]: @chunfuwen , line 158 is different to line 160. For line 158, rename the guest image file will make guest fail to start, so need set ignore_status to True.

File: libvirt/tests/src/virsh_cmd/monitor/virsh_domstate.py [hunk 2]
```diff
@@ -259,6 +291,8 @@ def run(test, params, env):
     finally:
         qemu_conf.restore()
         libvirtd.restart()
+        if vm_action == "start" and start_action == "rename":
+            os.rename(new_image_source, image_source)
         vm.destroy(gracefully=False)
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/monitor/virsh_domstate.py]: Better add judegement whether vm is alive before destroy it

File: libvirt/tests/src/virsh_cmd/monitor/virsh_domstate.py (hunk 1 above)
Comment (kylazhang) [reply libvirt/tests/src/virsh_cmd/monitor/virsh_domstate.py]: please add comments why need start fail here, thanks

**Pull request merged**